tests/testthat/test-assert-that.R

```diff
@@ -0,0 +1,14 @@
+context("assert_that")
+
+test_that("assert_that handles long false assertions gracefully", {
+    expect_error(
+        assert_that(isTRUE(10 + sqrt(25) + sum(1:10) + sqrt(25) + sum(11:20) + sqrt(25) + sum(21:30) + sqrt(25) + sum(31:40) + sqrt(25) + sum(41:50))),
+        "^isTRUE\\(x = 10 \\+ sqrt\\(25\\) \\+ sum\\(1\\:10\\) \\+ sqrt\\(25\\) \\+ sum\\(11\\:20\\) \\+ \\.\\.\\. is not TRUE$"
```

Comment:
The point of the regular expression is to make sure that you've got the right error message, not to exactly match the text. To that end, I'd suggest just matching `[.]{3}`.

Reply:
I disagree, actually. This test as it's written would catch a change that accidentally suppresses the first part of the error message and returns a NULL (which can happen silently in lots of settings).

A test with `[.]{3}` would pass if that happened and the error message was just `... is not TRUE`

Reply:
Then make it a little more specific; your current test is too fragile.

Reply:
ok sure. Changed to `"^isTRUE\\(.* [.]{3} is not TRUE$"` to catch the idea that at least some of the code passed in the expression is returned in the result